### ISSUE TITLE
Bug: ghost compose in `test_ldmcu`

### ISSUE THREAD

**Issue by MattePalte** [qclib/qclib#139]
# Environment

- **qiskit.__version__**: 0.23.2
- **qclib version**: b4c03af2a4ea585e86aeae1f9851ca953a2e9192 (latest main)
- **Python version**: 3.8.0
- **Operating system**: Ubuntu 18.04

# What is happening?
In the test file [test_ldmcu.py](https://github.com/qclib/qclib/blob/master/test/gates/test_ldmcu.py#L85) in the function `test_linear_toffoli2` we query the state of `circuit2` via `qclib.util.get_state(circuit2)`, but the circuit doesn't have the `mc_gate` in it (as we would expect after since we have instantiated it).
**NOTE**: This problem is present also in the next two tests: `test_linear_toffoli1`.

# How can we reproduce the issue?
I inserted some prints statements to see the state of the circuit:
```python
def test_linear_toffoli2(self):
    """ Testing Toffoli control 110"""
    gate_x = np.array([[0, 1], [1, 0]])

    controls2 = QuantumRegister(3)
    target2 = QuantumRegister(1)
    circuit2 = QuantumCircuit(controls2, target2)

    circuit2 = qiskit.QuantumCircuit(4)
    circuit2.x(2)
    circuit2.x(3)
    circuit2.x(0)
    state1 = qclib.util.get_state(circuit2)

    controls1 = QuantumRegister(3)
    target1 = QuantumRegister(1)
    circuit1 = qiskit.QuantumCircuit(controls1, target1)

    Ldmcu.ldmcu(circuit1, gate_x, controls1, target1, ctrl_state='110')

    print("Before Compose:")
    print(circuit2.draw())
    circuit2.compose(circuit1, circuit2.qubits)
    print("After Compose:")
    print(circuit2.draw())

    state2 = qclib.util.get_state(circuit2)

    print("Expected:")
    # what it should be
    # the following line is not in the code
    qc = circuit2.compose(circuit1, circuit2.qubits)
    print(qc.draw())
    self.assertTrue(np.allclose(state1, state2))
```
Output:
```
Before Compose:
     ┌───┐
q_0: ┤ X ├
     └───┘
q_1: ─────
     ┌───┐
q_2: ┤ X ├
     ├───┤
q_3: ┤ X ├
     └───┘
After Compose:
     ┌───┐
q_0: ┤ X ├
     └───┘
q_1: ─────
     ┌───┐
q_2: ┤ X ├
     ├───┤
q_3: ┤ X ├
     └───┘
Expected:
     ┌───┐┌────────┐
q_0: ┤ X ├┤0       ├
     └───┘│        │
q_1: ─────┤1       ├
     ┌───┐│  Ldmcu │
q_2: ┤ X ├┤2       ├
     ├───┤│        │
q_3: ┤ X ├┤3       ├
     └───┘└────────┘
```

# What should happen?
I would have expected the `circuit2` to use the `Ldmcu`, but as we can see the compose is ineffective, and the circuit doesn't have the `Ldmcu` in it.
The problem is in the use of the API `compose()`. See [here](https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.compose.html).

# Any suggestions?
The correct way to use it is:
```python
qc = circuit2.compose(circuit1, circuit2.qubits)
# or
circuit2.compose(circuit1, circuit2.qubits, inplace=True)
```

**Comment by israelferrazaraujo** [comment]
Hi @MattePalte,
You make an excellent point! Thank you for taking the time to provide such a detailed explanation.

**Comment by israelferrazaraujo** [comment]
Fixed! PR #140.
